storage/grpc_client.go

```diff
@@ -1042,6 +1043,16 @@ func (c *grpcStorageClient) NewRangeReader(ctx context.Context, params *newRange
 	// This is the size of the entire object, even if only a range was requested.
 	size := obj.GetSize()
 
+	// Only support checksums when reading an entire object, not a range.
+	var (
+		wantCRC  uint32
+		checkCRC bool
+	)
+	if checksums := msg.GetObjectChecksums(); checksums != nil && checksums.Crc32C != nil && params.offset == 0 && params.length < 0 {
```

Comment:
I don't have the full context here, but do you really want params.length to be negative here?

Reply:
I believe this is so that we only check the checksums on the entire object read, in which case params.length would be negative to indicate no limit.

Reply:
Yes, Brenna is correct.